Integrated bight tox unification changes

addison-grant opened the pull request:
This is a merge of #29, #25, #30, #31, #32, and #33, all on top of `staging`.

This draft is mainly for convenient viewing and testing of all the other Pull Requests combined. We could merge this and close the other pull requests if the changes are satisfactory.

addison-grant committed:
Properly separates dataset metadata

When referencing the summary data directly, the grouping variables are not properly taken into account. Use pick(everything()) instead (cur_data has been deprecated).

Update writelog statements to match code changes on this branch.

addison-grant committed:
Modify qa code collapsing

Instead of collapsing a group of QA Codes like this
"AO" "A" "X, C"

into
"AO; A; X, C"

Darrin suggested we should collapse that into the unique codes separated by commas, so for the above example, we would produce this:
"A, C, O, X"

addison-grant committed:
Update comment collapsing

Suppose you have a comments group like this:
```
> comments
[1] "asd" NA    ""    " "   "efg"
```

We want it to collapse to just
```
[1] "asd; efg"
```

We can do that with something like this:
```R
library(stringr) # str_trim, str_flatten
library(purrr) # discard
library(dplyr) # na_if
comments |>
  str_trim() |>
  na_if("") |>
  discard(is.na) |>
  str_flatten(collapse="; ")
```

addison-grant committed:
Handle all NA cases

In 1998-2013 results data, there are some groups that have result or control with all NA values. To handle that case, we can filter out those batches.

addison-grant committed:
Write log

addison-grant committed:
Update log writing

addison-grant committed:
Add tibble package for is_tibble function

addison-grant committed:
Fix variable name

addison-grant committed:
Update write log

addison-grant committed:
Optionally group by dilution

Dilution only appears in 1994, and if it does appear, I think we want to group by it

addison-grant committed:
Undo ifelse split

The change wasn't intentional

addison-grant committed:
Leave dilution NA if not in input

This assumes that the input results data doesn't have any values for dilution except for positive integers representing percentages. The input dilution percentages should be NA otherwise.

addison-grant committed:
Handle case where column exists but values NA

Previously NA values would be converted to characters.

addison-grant committed:
Merge branch '24-toxsummary-function-metadata-issue' into bight-tox-integration

addison-grant committed:
Merge branch '26-qa-code-modification-sort-unique' into bight-tox-integration

addison-grant committed:
Merge branch '27-handle-empty-comments-better' into bight-tox-integration

addison-grant committed:
Merge branch 'dilution--fix-for-1994' into bight-tox-integration

addison-grant committed:
Merge branch 'treatment--fix-na-as-such' into bight-tox-integration

addison-grant committed:
Check if vector instead of tibble

I realized that dplyr::pull returns a vector, not a tibble

addison-grant committed:
Update writelog for is.vector check

addison-grant committed:
Merge remote-tracking branch 'origin/28-bad-batches-all-na-results-andor-control' into bight-tox-integration

addison-grant committed:
Revert "Add tibble package for is_tibble function"

This reverts commit 16420421e0d3ee2af5ea1082b58ec1bfd6cfb440.

addison-grant committed:
Correct comment for change

I copied and pasted the writelog section, but forgot to update the
log text.

r7butler commented:
Looks good, but I'm surprised to not see updates to the roxygen comments, given that we are using new functions which I assume come from certain libraries, and we'll want to make sure they are imported at the top of the function definitions where they get used

I think we will want to ensure that the testing was done with resetting the R session and clearing variables in the environment

Sometimes in R, stuff will work because it will see stuff in the namespace left over from other times you ran stuff, and when you run it on another machine for the first time, it breaks

I'd expect that if we run devtools::document() that we would also see additional files created

We can go through this together next time you are in, it shouldnt take too long

addison-grant committed:
Fix group by to incorporate dilution if available

This is already utilized below in the main summary group_by. I forgot to add it here.

addison-grant commented:
I ran `devtools::document`, and noticed a couple changes that don't seem related to this branch. Maybe we can look at them when you're available.

As far as other functions I used, I believe they are all part of base, dplyr, stringr, or purr, and the last three are already all in the `DESCRIPTION` file.

https://github.com/SCCWRP/SQOUnified/blob/e3815b61e6f3cbd1167d207515d9230af939c0f0/DESCRIPTION#L35-L38

addison-grant committed:
Be more consistent with if and else vs. ifelse

addison-grant commented:
Here is the diff from `devtools::document`

```diff
diff --git a/DESCRIPTION b/DESCRIPTION
index 820ed9c..572634e 100644
--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -43,4 +43,4 @@ Imports:
   rmarkdown
 Depends: 
   R (>= 3.5.0)
-RoxygenNote: 7.2.3
+RoxygenNote: 7.3.2
diff --git a/man/SQOUnified.Rd b/man/SQOUnified.Rd
index 74358d8..7badfe6 100644
--- a/man/SQOUnified.Rd
+++ b/man/SQOUnified.Rd
@@ -12,7 +12,8 @@ SQOUnified(
   tox = NULL,
   logfile = file.path(getwd(), "logs", format(Sys.time(), "\%Y-\%m-\%d_\%H-\%M-\%S"),
     "log.Rmd"),
-  verbose = F
+  verbose = F,
+  knitlog = F
 )
 }
 \arguments{
diff --git a/man/SoCalRivpacs.Rd b/man/SoCalRivpacs.Rd
index 08f2383..ca51271 100644
--- a/man/SoCalRivpacs.Rd
+++ b/man/SoCalRivpacs.Rd
@@ -16,7 +16,8 @@ SoCalRivpacs(
   observed.taxa = socal.example.taxa,
   logfile = file.path(getwd(), "logs", format(Sys.time(), "\%Y-\%m-\%d_\%H-\%M-\%S"),
     "log.txt"),
-  verbose = F
+  verbose = F,
+  knitlog = F
 )
 }
 \description{
```

addison-grant committed:
Add string unique to comment concatenation

addison-grant committed:
Commit minor devtools::document output

addison-grant committed:
Put NA (character type) for empty comments

addison-grant committed:
Rename a few variables

These names better match what the variables represent.

addison-grant committed:
Fix out-of-date comment

addison-grant committed:
Gitignore .DS_Store files

addison-grant committed:
Do not assign SQO category for control samples

addison-grant committed:
Handle cases where labrep counts are not equal

Some of the result samples might be NA or some of the control samples might be NA. The inner join would cause cutting out some of one or the other type of samples.

addison-grant committed:
Update writelog calls with code changes

addison-grant committed:
Fix comment quotation marks

addison-grant committed:
Only include QA code "A" results in calculations

QA Code "A" refers to results acceptable for analysis. "X" indicates excluded from analysis.

addison-grant committed:
Change default result sampletype to "Result"

This was discussed with Alvina, Paul, and Robert.

The term "Result" is more appropriate for analysis.

addison-grant committed:
Revert "Only include QA code "A" results in calculations"

This reverts commit 072a1b30e58042a8c6fe221792438a8b89507731.

addison-grant committed:
Fix comment about Grab vs Result sample type

addison-grant committed:
Handle certain NaN, Inf, and divide by 0 cases
